api/api.go

```diff
@@ -76,225 +78,286 @@ func (api *ImportAPI) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 }
 
 func (api *ImportAPI) addJob(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
 	job, err := models.CreateJob(r.Body)
 	defer r.Body.Close()
 	if err != nil {
-		log.Error(err, nil)
+		log.ErrorCtx(ctx, errors.WithMessage(err, "api endpoint addJob error - Bad client request received"), nil)
 		http.Error(w, "Bad client request received", http.StatusBadRequest)
 		return
 	}
 
 	logData := log.Data{"recipeID": job.RecipeID}
 	auditParams := common.Params{"recipeID": job.RecipeID}
-	if err := api.auditor.Record(r.Context(), addJobAction, actionAttempted, auditParams); err != nil {
-		handleAuditError(w, err, logData)
+	if auditError := api.auditor.Record(ctx, addJobAction, actionAttempted, auditParams); auditError != nil {
+		handleAuditError(ctx, w, addJobAction, actionAttempted, auditError, logData)
 		return
 	}
 
-	createdJob, err := api.jobService.CreateJob(r.Context(), job)
-	if err != nil {
-		if err := api.auditor.Record(r.Context(), addJobAction, actionUnsuccessful, auditParams); err != nil {
-			handleAuditError(w, err, logData)
-			return
+	var createdJob *models.Job
+	b, err := func() (bytes []byte, err error) {
+		createdJob, err = api.jobService.CreateJob(ctx, job)
+		if err != nil {
+			log.ErrorCtx(ctx, errors.WithMessage(err, "addJob endpoint: error creating job resource"), logData)
+			return nil, err
 		}
-		setErrorCode(w, err)
-		return
-	}
 
-	logData["job"] = job
-	auditParams["createdJobID"] = createdJob.ID
+		logData["job"] = createdJob
+		auditParams["createdJobID"] = createdJob.ID
+
+		bytes, err = json.Marshal(createdJob)
+		if err != nil {
+			log.ErrorCtx(ctx, errors.WithMessage(err, "addJob endpoint: failed to marshal job resource into bytes"), logData)
+			return nil, err
```

Comment:
If you are going to name the variables in the function signature, then no need to return the arguments, i.e instead of:

```go
return nil, err
```

just:

```go
return
```